openlibrary/core/bookshelves_events.py

```diff
@@ -124,6 +140,23 @@ def select_distinct_by_user_type_and_year(cls, username, event_type, year):
 
         return list(oldb.query(query, vars=data))
 
+    @classmethod
+    def total_yearly_reading_goals(cls, since: date | None = None) -> int:
+        """Returns a Storage object of <Storage {'count': int}> where 'count' specifies the
+        number reading goals updated. `since` may be used
+        to limit the result to those reading goals updated since a specific
+        date. Any python datetime.date type should work.
+        :param since: returns all reading goals after date
+        """
+        oldb = db.get_db()
+
+        query = "SELECT count(*) from yearly_reading_goals"
```

Comment:
After moving this function to `core/yearly_reading_goals.py`, please replace the hard-coded table name with `TABLENAME`: https://github.com/internetarchive/openlibrary/blob/112cf1468a0ead2f2e90294596c7608dcb58c43f/openlibrary/core/yearly_reading_goals.py#L6